clients/fides-js/src/lib/events.ts

```diff
@@ -0,0 +1,52 @@
+import { FidesCookie } from "./cookie";
+
+/**
+ * Defines the available event names:
+ * - FidesInitialized: dispatched when initialization is complete, from Fides.init()
+ * - FidesUpdated: dispatched when preferences are updated, from updateConsentPreferences() or Fides.init()
+ */
+export type FidesEventType = "FidesInitialized" | "FidesUpdated";
+
+// Bonus points: update the WindowEventMap interface with our custom event types
+declare global {
+  interface WindowEventMap {
+    FidesInitialized: FidesEvent;
+    FidesUpdated: FidesEvent;
+  }
+}
+
+/**
+ * Defines the properties available on event.detail. As of now, these are an
+ * explicit subset of properties from the Fides cookie
+ * TODO: add identity and meta?
```

Comment:
Decide on this -  I think it makes sense to include these in the event, I could see a listener wanting a bit of information about the identity when processing the preferences, etc.

Reply:
Does gtm need identity or other metadata?

Reply:
Nothing needs it _yet_, but I'm wondering what a developer might need when integrating consent into their app via these events...